src/Microsoft.ML.LightGBM/WrappedLightGbmBooster.cs

```diff
@@ -96,10 +96,24 @@ internal unsafe string GetModelString()
 
         private static double[] Str2DoubleArray(string str, char delimiter)
         {
-            return str.Split(delimiter).Select(
-                x => { double t; double.TryParse(x, out t); return t; }
-            ).ToArray();
+            var values = new List<double>();
+            foreach (var token in str.Split(delimiter))
+            {
+                var trimmed = token.Trim();
 
+                if (trimmed.Equals("inf", StringComparison.OrdinalIgnoreCase))
```

Comment:
Does this really work?  The strings output by LightGBM are +nan.0, +inf.0, and -inf.0.  How are you handling the trailing .0 bit?